src/utf8.c

```diff
@@ -47,6 +52,10 @@ utf32_to_utf8(uint32_t unichar, char *buffer)
         length = 2;
         head = 0xc0;
     }
+    /* Handle surrogates */
+    else if (0xd800 <= unichar && unichar <= 0xdfff) {
+        goto ill_formed_code_unit_subsequence;
```

Comment:
goto from an if branch to an else branch is too wacky for me :) I suggest one of the following:

1. Repeat the `buffer[0] = '\0'; return 0;` code.
2. Move `ill_formed_code_unit_subsequence` label to the end of the function, after the `return length + 1;`, and `goto` there from both branches.

Reply:
Fixed using 2nd option